src/glean/private/ping.h

```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef PING_H
+#define PING_H
+
+#include "vpnglean.h"
+
+class Ping final {
+ public:
+  constexpr explicit Ping(int aId) : m_id(aId) {}
+
+  void submit() const { glean_submit_ping_by_id(m_id); };
+
```

Comment:
No `testBeforeNextSubmit` support? (Not yet?)

Reply:
Heh, I forgot about it 👀 I'll do it in a follow up this should be enough I think. My next PR will actually record events and test the recordings, we will see if I miss test_before_next_submit. For now: https://mozilla-hub.atlassian.net/browse/VPN-3338